File: app_canibuythat/src/main/java/com/gb/canibuythat/ui/InputDialog.java
package com.gb.canibuythat.ui;

import android.os.Bundle;
import android.support.annotation.StringRes;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

public class InputDialog extends BaseDialogFragment {

    private static final String EXTRA_TITLE = "title";
    private static final String EXTRA_CURRENT_INPUT = "current_input";

    private @StringRes int btnStringResId;
    private View.OnClickListener onClickListener;

    public static InputDialog newInstance(String title, String currentInput) {
        InputDialog inputDialog = new InputDialog();

        Bundle args = new Bundle();
        args.putString(EXTRA_TITLE, title);
        args.putString(EXTRA_CURRENT_INPUT, currentInput);
        inputDialog.setArguments(args);

        return inputDialog;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = super.onCreateView(inflater, container, savedInstanceState);

        Bundle b = getArguments();

        setTitle(b.getString(EXTRA_TITLE));
        input.setText(b.getString(EXTRA_CURRENT_INPUT));
        input.setVisibility(View.VISIBLE);

        super.setPositiveButton(btnStringResId, v -> {
            dismiss();
            if (onClickListener != null) {
                onClickListener.onClick(v);
            }
        });
        return view;
    }

    @Override
    public InputDialog setPositiveButton(int buttonTextId, View.OnClickListener listener) {
        btnStringResId = buttonTextId;
        onClickListener = listener;
        return this;
    }

    public String getInput() {
        return input.getText().toString();
    }
}

File: app/src/main/java/com/gb/canibuyit/ui/InputDialog.kt
package com.gb.canibuyit.ui

import android.os.Bundle
import android.support.annotation.StringRes
import android.view.View
import com.gb.canibuyit.util.show
import kotlinx.android.synthetic.main.prompt_dialog_layout.*

class InputDialog : BaseDialogFragment() {

    @StringRes
    private var btnStringResId: Int = 0
    private var onClickListener: ((View) -> Unit)? = null

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        arguments?.getString(EXTRA_TITLE)?.let { setTitle(it) }
        arguments?.getString(EXTRA_CURRENT_INPUT)?.let { text_input.setText(it) }
        text_input.show()

        super.setPositiveButton(btnStringResId) {
            dismiss()
            onClickListener?.invoke(it)
        }
    }

    override fun setPositiveButton(buttonTextId: Int, listener: ((View) -> Unit)?): InputDialog {
        btnStringResId = buttonTextId
        onClickListener = listener
        return this
    }

    val input: String
        get() = text_input.text.toString()

    companion object {
        const val EXTRA_TITLE = "title"
        const val EXTRA_CURRENT_INPUT = "current_input"

        fun newInstance(title: String, currentInput: String?) = InputDialog().apply {
            arguments = Bundle().apply {
                putString(EXTRA_TITLE, title)
                putString(EXTRA_CURRENT_INPUT, currentInput)
            }
        }
    }
}
